src/interfaces.ts

```diff
@@ -0,0 +1,213 @@
+    // a JWT token if ever found as the return type of a procedure. Can be of the
+    // form: `my_schema.my_type`. You may use quotes as needed:
+    // `"my-special-schema".my_type`.
+    jwtPgTypeIdentifier?: string,
+    // [DEPRECATED] The audience to use when verifing the JWT token. Deprecated,
+    // use `jwtVerifyOptions.audience` instead.
+    /* @middlewareOnly */
+    jwtAudiences?: Array<string>,
+    // Some one-to-one relations were previously detected as one-to-many - should
+    // we export 'only' the old relation shapes, both new and old but mark the
+    // old ones as 'deprecated', or 'omit' the old relation shapes entirely
+    legacyRelations?: 'only' | 'deprecated' | 'omit',
+    // ONLY use this option if you require the v3 typenames 'Json' and 'Uuid'
+    // over 'JSON' and 'UUID'
+    legacyJsonUuid?: boolean,
+    // Turns off GraphQL query logging. By default PostGraphile will log every
+    // GraphQL query it processes along with some other information. Set this to
+    // `true` to disable that feature.
+    /* @middlewareOnly */
+    disableQueryLog?: boolean,
+    // A plain object specifying custom config values to set in the PostgreSQL
+    // transaction (accessed via `current_setting('my.custom.setting')`) or a
+    // function which will return the same (or a Promise to the same) based on
+    // the incoming web request (e.g. to extract session data)
+    /* @middlewareOnly */
+    pgSettings?: { [key: string]: mixed } | ((req: IncomingMessage) => Promise<{ [key: string]: mixed }>),
+    // Some graphile-build plugins may need additional information available on
+    // the `context` argument to the resolver - you can use this function to
+    // provide such information based on the incoming request - you can even use
+    // this to change the response [experimental], e.g. setting cookies
+    /* @middlewareOnly */
+    additionalGraphQLContextFromRequest?: (req: IncomingMessage, res: ServerResponse) => Promise<{}>,
+    // [experimental] Plugin hook function, enables functionality within
+    // PostGraphile to be expanded with plugins. Generate with
+    // `makePluginHook(plugins)` passing a list of plugin objects.
+    /* @middlewareOnly */
+    pluginHook?: PluginHookFn,
+    // Should we use relay pagination, or simple collections?
+    // "omit" (default) - relay connections only,
+    // "only" - simple collections only (no Relay connections),
+    // "both" - both
+    simpleCollections?: 'omit' | 'both' | 'only',
+  }
+
+  /**
+   * A request handler for one of many different `http` frameworks.
+   */
+  export interface HttpRequestHandler {
+    (req: IncomingMessage, res: ServerResponse, next?: (error?: mixed) => void): void
+    (ctx: { req: IncomingMessage, res: ServerResponse }, next: () => void): Promise<void>
+    getGraphQLSchema: () => Promise<GraphQLSchema>
+    formatError: (e: GraphQLError) => GraphQLError
+    pgPool: Pool
```

Comment:
I think this has more stuff on it now, I should check.

Reply:
https://github.com/graphile/postgraphile/blob/master/src/postgraphile/http/createPostGraphileHttpRequestHandler.js#L813-L816

Reply:
20a50fe 

this one was a bit of a drag, inducing a little more typing refactor than I had wanted.  ill see if i can annotate it.  i bundled your remark from https://github.com/graphile/postgraphile/pull/794/files#r200834248 in this commit too.